packages/slice-machine/src/domain/customType.ts

```diff
@@ -129,6 +129,28 @@ export function getSectionSliceZoneConfig(
   return maybeSliceZone?.config ?? undefined;
 }
 
+// TODO add tests
+// We block user from deleteing section containing UID field,
+// so we can expect uid to be in the main section
+export function getUidField(customType: CustomType): UID | undefined {
+  const mainSectionEntry = getMainSectionEntry(customType);
+  const [_, mainSection] = mainSectionEntry ?? [];
+
+  return Object.values(mainSection ?? {}).find(
+    (field): field is UID => field.type === "UID",
+  );
+}
+
+// TODO add tests
+export function findSectionWithUidField(customType: CustomType): string | null {
```

Comment:
❓#question

I'm lost here, you're saying that we expect UID to be in the main section but here you have another function to find the section with the UID field. One of the two think is strange then. Should we not use that management in the previous one?

Also maybe if we sure it's always the main section, we don't care to look for the UID field in other section. Can it be other sections?

Reply:
yes, you're right, since the Main section has an uid field, and we will prevent user form deleting that section, we shouldn't need to look for uid in other ones

Reply:
As discussed in private, old version of SM may not have the UID in the main section

Reply:
changed the logic to find the UID wherever it is, 
also after chat with Benjamin, added option to add uid via the new button if for any reason it's missing in all sections